jdaviz/core/marks.py

```diff
@@ -104,8 +104,17 @@ def set_y_unit(self, unit=None):
                 return
             unit = self.viewer.state.y_display_unit
         unit = u.Unit(unit)
+
         if self.yunit is not None:
-            self.y = (self.y * self.yunit).to_value(unit)
+            if self.viewer.default_class is Spectrum1D:
```

Comment:
@kecnry Think this is better?

Reply:
yeah, I actually think that makes a lot of sense, since it then uses `get_object(cls=Spectrum1D)`!